terraform/cloudwatch.tf.json.template.py

```diff
@@ -329,10 +329,20 @@ def dashboard_body() -> str:
         ),
         {
             'aws_cloudwatch_dashboard': {
-                'dashboard': {
-                    'dashboard_name': config.qualified_resource_name('dashboard'),
-                    'dashboard_body': dashboard_body()
-                }
+                'indexer': {
+                    'dashboard_name': config.qualified_resource_name('indexer'),
+                    'dashboard_body': dashboard_body('indexer')
+                },
+                **(
+                    {
+                        'mirror': {
+                            'dashboard_name': config.qualified_resource_name('mirror'),
+                            'dashboard_body': dashboard_body('mirror')
+                        }
+                    }
+                    if config.enable_mirroring else
+                    {}
+                )
```

Comment:
```suggestion
                **iif(config.enable_mirroring, {
                    'mirror': {
                        'dashboard_name': config.qualified_resource_name('mirror'),
                        'dashboard_body': dashboard_body('mirror')
                    }
                }),
```

Reply:
We should probably be using this function anywhere the `else` value is an empty/default value.

Reply:
`iif()` can't be used here since it is not lazy, causing `dashboard_body('mirror')` to be called even if `config.enable_mirroring` is False, resulting in an AssertionError ('mirror' not in valid_names)